### ISSUE TITLE
Support TLS.FrontendValidation within the Gateway spec

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#3316]
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

Support the new Client Cert Validation field introduced by upstream
https://gateway-api.sigs.k8s.io/geps/gep-91/

Its the same feature as https://github.com/envoyproxy/gateway/issues/88, and has been implemented in the [project using ClientTrafficPolicy](https://gateway.envoyproxy.io/v1.0.1/api/extension_types/#clientvalidationcontext), so the translation can be reused

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by arkodg** [comment]
blocked on https://github.com/envoyproxy/gateway/issues/3265

**Comment by zufardhiyaulhaq** [comment]
I am interested in working on this feature.
want to understand which one to prioritize if gateway API enable but we are not enabling it on ClientTraficPolicy?

**Comment by zufardhiyaulhaq** [comment]
let me finish #3202 before working on this one

**Comment by arkodg** [comment]
> I am interested in working on this feature. want to understand which one to prioritize if gateway API enable but we are not enabling it on ClientTraficPolicy?

@zufardhiyaulhaq we'll need to simultaneously support both fields, until there is complete feature parity in upstream

**Comment by zufardhiyaulhaq** [comment]
@arkodg @shawnh2 after checking the spec https://gateway-api.sigs.k8s.io/geps/gep-91/
seems there are no optional features there. which means that if ClientTrafficPolicy set the optional to true, we need to rewrite it to true right?

**Comment by zufardhiyaulhaq** [comment]
@arkodg 
```
@zufardhiyaulhaq we'll need to simultaneously support both fields, until there is complete feature parity in upstream
```

which one to be used if different secret is being used on Gateway object & ClientTrafficPolicy object? do we need to merge the certificate?

**Comment by arkodg** [comment]
yah lets merge/append the `caCerts` , this will help with migration in the future

**Comment by 3deep5me** [comment]
Any news on this topic?

**Comment by zirain** [comment]
need this for following conformance tests:
- tests.GatewayInvalidTLSBackendConfiguration
- tests.GatewayTLSBackendClientCertificate
- tests.GatewayFrontendClientCertificateValidation
- tests.GatewayInvalidFrontendClientCertificateValidation
- tests.GatewayFrontendInvalidDefaultClientCertificateValidation
- tests.GatewayFrontendClientCertificateValidationInsecureFallback

**Comment by rouke-broersma** [comment]
@zirain @arkodg since CTP is way more flexible than TLS.FrontendValidation and gateway-api does not currently seem willing to take the risk of introducing more flexibility, has envoy-gateway considered what the path moving forward would look like? 

> @zufardhiyaulhaq we'll need to simultaneously support both fields, until there is complete feature parity in upstream

As it currently stands this means supporting client certificate validation in CTP indefinitely as upstream has chosen to not support the mitigation against HTTP2 that envoy-gateway has chosen (allow but disable HTTP2).

**Comment by arkodg** [comment]
yeah, we have no plans to remove ClientValidation support from CTP, the Gateway API Extensions are designed to add this richer feature API surface than upstream
